db/migrate/20170529142557_unify_container_definition_and_container.rb

```diff
@@ -0,0 +1,159 @@
+class UnifyContainerDefinitionAndContainer < ActiveRecord::Migration[5.0]
+  class ContainerDefinition < ActiveRecord::Base
+    has_many :container_port_configs
+    has_many :container_env_vars
+    has_one :security_context, :as => :resource
+    has_one :container
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Container < ActiveRecord::Base
+    belongs_to :container_definition
+    self.inheritance_column = :_type_disabled
+  end
+
+  class SecurityContext < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class ContainerEnvVar < ActiveRecord::Base
+    belongs_to :container_definition
+    self.inheritance_column = :_type_disabled
+  end
+
+  class ContainerPortConfig < ActiveRecord::Base
+    belongs_to :container_definition
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    # attributes
+    add_column :containers, :image,              :string
+    add_column :containers, :image_pull_policy,  :string
+    add_column :containers, :memory,             :string
+    add_column :containers, :cpu_cores,          :float
+    add_column :containers, :container_group_id, :bigint
+    add_column :containers, :privileged,         :boolean
+    add_column :containers, :run_as_user,        :bigint
+    add_column :containers, :run_as_non_root,    :boolean
+    add_column :containers, :capabilities_add,   :string
+    add_column :containers, :capabilities_drop,  :string
+    add_column :containers, :command,            :text
+
+    say_with_time("Copying over columns from container_definition to container") do
+      Container.all.each do |container|
+        container_definition = ContainerDefinition.find(container.container_definition_id)
```

Comment:
I dont think so. in order to have containers at all the inventory refresh has to go through and save provider->pod->container_def->container.